radiation/python/radsw/stencils_sw_gt4py.py

```diff
@@ -0,0 +1,4107 @@
+    eps1,
+    eps,
+    nuvb,
+    NG16,
+    NG17,
+    NG18,
+    NG19,
+    NG20,
+    NG21,
+    NG22,
+    NG23,
+    NG24,
+    NG25,
+    NG26,
+    NG27,
+    NG28,
+    NG29,
+    NS16,
+    NS17,
+    NS18,
+    NS19,
+    NS20,
+    NS21,
+    NS22,
+    NS23,
+    NS24,
+    NS25,
+    NS26,
+    NS27,
+    NS28,
+    NS29,
+)
+
+rebuild = False
+validate = True
+backend = "gtc:gt:cpu_ifirst"
+
+# Eventually these will be provided by rad_initialize
+isubcsw = 2
+lhsw0 = True
+lhswb = False
+lflxprf = False
+lfdncmp = True
+
+
+@stencil(
+    backend=backend,
+    rebuild=rebuild,
+    externals={
+        "s0": s0,
+        "con_g": con_g,
+        "con_avgd": con_avgd,
+        "con_amd": con_amd,
+        "con_amw": con_amw,
+        "amdw": amdw,
+        "amdo3": amdo3,
+        "iswrgas": iswrgas,
+        "iswcliq": iswcliq,
+        "iovrsw": iovrsw,
+        "nbdsw": nbdsw,
+        "ftiny": ftiny,
+        "oneminus": oneminus,
+    },
+)
+def firstloop(
+    plyr: FIELD_FLT,
+    plvl: FIELD_FLT,
+    tlyr: FIELD_FLT,
+    tlvl: FIELD_FLT,
+    qlyr: FIELD_FLT,
+    olyr: FIELD_FLT,
+    gasvmr: Field[type_10],
+    clouds: Field[type_9],
+    aerosols: Field[(DTYPE_FLT, (nbdsw, 3))],
+    sfcalb: Field[(DTYPE_FLT, (4,))],
+    dzlyr: FIELD_FLT,
+    delpin: FIELD_FLT,
+    de_lgth: FIELD_2D,
+    cosz: FIELD_2D,
+    idxday: FIELD_2DBOOL,
+    solcon: float,
+    cosz1: FIELD_2D,
+    sntz1: FIELD_2D,
+    ssolar: FIELD_2D,
+    albbm: Field[(DTYPE_FLT, (2,))],
+    albdf: Field[(DTYPE_FLT, (2,))],
+    tem1: FIELD_FLT,
+    tem2: FIELD_FLT,
+    pavel: FIELD_FLT,
+    tavel: FIELD_FLT,
+    h2ovmr: FIELD_FLT,
+    o3vmr: FIELD_FLT,
+    tem0: FIELD_FLT,
+    coldry: FIELD_FLT,
+    temcol: FIELD_FLT,
+    colamt: Field[type_maxgas],
+    colmol: FIELD_FLT,
+    tauae: Field[type_nbdsw],
+    ssaae: Field[type_nbdsw],
+    asyae: Field[type_nbdsw],
+    cfrac: FIELD_FLT,
+    cliqp: FIELD_FLT,
+    reliq: FIELD_FLT,
+    cicep: FIELD_FLT,
+    reice: FIELD_FLT,
+    cdat1: FIELD_FLT,
+    cdat2: FIELD_FLT,
+    cdat3: FIELD_FLT,
+    cdat4: FIELD_FLT,
+    zcf0: FIELD_2D,
+    zcf1: FIELD_2D,
+):
+    from __externals__ import (
+        s0,
+        con_g,
+        con_avgd,
+        con_amd,
+        con_amw,
+        amdw,
+        amdo3,
+        iswrgas,
+        iswcliq,
+        iovrsw,
+        nbdsw,
+        ftiny,
+        oneminus,
+    )
+
+    with computation(FORWARD), interval(0, 1):
+
+        s0fac = solcon / s0
+
+        if idxday:
+            cosz1 = cosz
+            sntz1 = 1.0 / cosz
+            ssolar = s0fac * cosz
+
+            # Prepare surface albedo: bm,df - dir,dif; 1,2 - nir,uvv.
+            albbm[0, 0, 0][0] = sfcalb[0, 0, 0][0]
+            albdf[0, 0, 0][0] = sfcalb[0, 0, 0][1]
+            albbm[0, 0, 0][1] = sfcalb[0, 0, 0][2]
+            albdf[0, 0, 0][1] = sfcalb[0, 0, 0][3]
+
+            zcf0 = 1.0
+            zcf1 = 1.0
+
+    with computation(FORWARD), interval(1, None):
```

Comment:
can these two computations be combined, using two separate 'with intervals'?

Reply:
I think so, will do it